src/handler.rs

```diff
@@ -176,6 +184,24 @@ impl RequestHandler {
                 return Ok(resp);
             }
 
+            // Metrics endpoint check
+            if metrics_request {
```

Comment:
Note that I moved `use prometheus::Encoder;` down here.

```suggestion
            #[cfg(linux)]
            if metrics_request {
                use prometheus::Encoder;

```